packages/twenty-server/src/workspace/calendar/repositories/calendar-event/calendar-event.service.ts

```diff
@@ -0,0 +1,218 @@
+
+    await this.workspaceDataSourceService.executeRawQuery(
+      `DELETE FROM ${dataSourceSchema}."calendarEvent" WHERE "id" = ANY($1)`,
+      [calendarEventIds],
+      workspaceId,
+      transactionManager,
+    );
+  }
+
+  public async getNonAssociatedCalendarEventIdsPaginated(
+    limit: number,
+    offset: number,
+    workspaceId: string,
+    transactionManager?: EntityManager,
+  ): Promise<ObjectRecord<CalendarEventAttendeeObjectMetadata>[]> {
+    const dataSourceSchema =
+      this.workspaceDataSourceService.getSchemaName(workspaceId);
+
+    const nonAssociatedCalendarEvents =
+      await this.workspaceDataSourceService.executeRawQuery(
+        `SELECT m.id FROM ${dataSourceSchema}."calendarEvent" m
+        LEFT JOIN ${dataSourceSchema}."calendarChannelEventAssociation" ccea
+        ON m.id = ccea."calendarEventId"
+        WHERE ccea.id IS NULL
+        LIMIT $1 OFFSET $2`,
+        [limit, offset],
+        workspaceId,
+        transactionManager,
+      );
+
+    return nonAssociatedCalendarEvents.map(({ id }) => id);
+  }
+
+  public async getICalUIDCalendarEventIdMap(
+    iCalUIDs: string[],
+    workspaceId: string,
+    transactionManager?: EntityManager,
+  ): Promise<Map<string, string>> {
+    if (iCalUIDs.length === 0) {
+      return new Map();
+    }
+
+    const dataSourceSchema =
+      this.workspaceDataSourceService.getSchemaName(workspaceId);
+
+    const calendarEvents: {
+      id: string;
+      iCalUID: string;
+    }[] = await this.workspaceDataSourceService.executeRawQuery(
+      `SELECT id, "iCalUID" FROM ${dataSourceSchema}."calendarEvent" WHERE "iCalUID" = ANY($1)`,
+      [iCalUIDs],
+      workspaceId,
+      transactionManager,
+    );
+
+    const iCalUIDsCalendarEvnetIdsMap = new Map<string, string>();
+
+    calendarEvents.forEach((calendarEvent) => {
+      iCalUIDsCalendarEvnetIdsMap.set(calendarEvent.iCalUID, calendarEvent.id);
+    });
+
+    return iCalUIDsCalendarEvnetIdsMap;
+  }
+
+  public async saveCalendarEvents(
+    calendarEvents: CalendarEvent[],
+    workspaceId: string,
+    transactionManager?: EntityManager,
+  ): Promise<void> {
+    if (calendarEvents.length === 0) {
+      return;
+    }
+
+    const dataSourceSchema =
+      this.workspaceDataSourceService.getSchemaName(workspaceId);
+
+    const valuesString = valuesStringForBatchRawQuery(calendarEvents, 14);
+
+    const values = calendarEvents.flatMap((calendarEvent) => [
+      calendarEvent.id,
+      calendarEvent.title,
+      calendarEvent.isCanceled,
+      calendarEvent.isFullDay,
+      calendarEvent.startsAt,
+      calendarEvent.endsAt,
+      calendarEvent.externalCreatedAt,
+      calendarEvent.externalUpdatedAt,
+      calendarEvent.description,
+      calendarEvent.location,
+      calendarEvent.iCalUID,
+      calendarEvent.conferenceSolution,
+      calendarEvent.conferenceUri,
+      calendarEvent.recurringEventExternalId,
+    ]);
```

Comment:
If we're flattening all the keys, we can use `Object.keys`

Reply:
It has to be in the same order as in the query, also it's not all the keys here